### PR TITLE
Fixed breaking change updates.

### PR DESCRIPTION
**Pull request by ryankopf**
My other PRs have too much work in them, this one is far more focused.

This PR:

Fixes:
```
boot(mode: StartMode, environment: &Environment)
```

Was:
```
boot(mode: StartMode, environment: &str)
```

Fixes:
```
fn routes(_ctx: &loco_rs::prelude::AppContext) -> AppRoutes {
```

Was:
```
fn routes() -> AppRoutes {
```

I've put this into its own PR for ease of merging, and verified everything runs properly after this change is applied.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -130,14 +130,14 @@ Reminder: `loco --version` will give you the current Loco framework which your a
...
-* **Braking changes**: Adding `Hooks::boot` application. Migration steps:
+* **Breaking changes**: Adding `Hooks::boot` application. Migration steps:
     ```rust
     // Load boot::{create_app, BootResult, StartMode} from loco_rs lib
     // Load migration: use migration::Migrator; Only when using DB
     // Adding boot hook with the following code
     impl Hooks for App {
       ...
-      async fn boot(mode: StartMode, environment: &str) -> Result<BootResult> {
+      async fn boot(mode: StartMode, environment: &Environment) -> Result<BootResult> {
```
Comment (kaplanelad) [inline CHANGELOG.md:140]: Why was this changed made? you change docs on version `v0.1.8, with upcoming changes to the Environment variable in the next release.

If someone wishes to upgrade from v0.1.7 to v0.1.8, what considerations should they be aware of?"

File: docs-site/content/blog/frontend-website.md [hunk 2]
```diff
@@ -119,7 +119,7 @@ $ pnpm dev
 
 First, move all our rest api endpoint under `/api` prefix. for doing it go to `src/app.rs`. in `routes` hooks function add `.prefix("/api")` to the default routes. 
 ```rust
-fn routes() -> AppRoutes {
+fn routes(_ctx: &loco_rs::prelude::AppContext) -> AppRoutes {
```
Comment (kaplanelad) [inline docs-site/content/blog/frontend-website.md:122]: Please update it to loco_rs::app::AppContext. Ensure that you apply this change consistently across all relevant sections in your modifications

File: src/boot.rs [hunk 3]
```diff
@@ -70,6 +70,7 @@ pub async fn start(boot: BootResult) -> Result<()> {
             serve(router, &app_context.config).await?;
         }
         (Some(router), None) => {
+            println!("Config: {:?}", app_context.config);
```
Comment (kaplanelad) [inline src/boot.rs:73]: I believe it is not relevant here. 
In additionally, be cautious about printing/logging all the configuration details, as it may include sensitive data that you wouldn't want to expose in logs or prints.

File: src/controller/mod.rs [hunk 4]
```diff
@@ -13,6 +13,7 @@
 //!    app::{AppContext, Hooks},
 //!    boot::{create_app, BootResult, StartMode},
 //!    controller::AppRoutes,
+//!    environment::Environment,
```
Comment (kaplanelad) [inline src/controller/mod.rs:16]: Why? If you go three lines down, you'll already find this declaration.

File: starters/lightweight-service/src/app.rs [hunk 5]
```diff
@@ -3,6 +3,7 @@ use loco_rs::{
     app::{AppContext, Hooks},
     boot::{create_app, BootResult, StartMode},
     controller::AppRoutes,
+    environment::Environment,
```
Comment (kaplanelad) [inline starters/lightweight-service/src/app.rs:6]: Revert all your changes in the state template. 

The starter template should always be aligned with the released version and not include breaking changes.

for more information, you can start [readme](https://github.com/loco-rs/loco/blob/master/starters/README.md)

**Review by kaplanelad — CHANGES_REQUESTED** [review]

File: starters/lightweight-service/src/app.rs (hunk 5 above)
Comment (ryankopf) [reply starters/lightweight-service/src/app.rs:6]: Do you mind if I the following pattern for the starter templates?

* Put a TODO before all "breaking changes", this will let other developers know that if they are trying to work with the bleeding edge changes, they will be able to see what to change, and that we haven't forgotten an item
* Go ahead and add any imports that will be needed in the next release, if the change is backwards compatible

File: src/controller/mod.rs (hunk 4 above)
Comment (ryankopf) [reply src/controller/mod.rs:16]: Thank you, I have removed the duplicate and sorted them alphabetically, following the convention in other docs

File: src/boot.rs (hunk 3 above)
Comment (ryankopf) [reply src/boot.rs]: This was an artifact from debugging, it has been removed, thank you

File: docs-site/content/blog/frontend-website.md (hunk 2 above)
Comment (ryankopf) [reply docs-site/content/blog/frontend-website.md]: Will do! Now that I see the correct reference, I have modified to just &AppContext whenever &loco_rs::app::AppContext is already imported.

**Comment by ryankopf** [comment]
Looks like these changes are made
